design_pim/overview.rst

```diff
@@ -0,0 +1,322 @@
+
+Summary
++++++++
+
+We just created our first module and made it available for other part of the application. You can now use it by requirering ``my_locale_logger``.
+
+
+Create our first backbone view
+==============================
+
+Know that we know how to create a requirejs module, let's create our first Backbone view.
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/new_view.js
+    define(
+        ['backbone'],
+        function (Backbone) {
+            return Backbone.View.extend({
+
+            });
+        }
+    );
+
+
+This is the simplest view we can create using backbone. It will result into an empty div if we add it to the DOM.
+
+Let's add some more interesting stuff to it:
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/new_view.js
+    define(
+        ['backbone'],
+        function (Backbone) {
+            return Backbone.View.extend({
+                events: {
+                    'click': 'clicked'
+                },
+
+                render: function () {
+                    this.$el.html('<div>Hello world</div>')
+                },
+
+                clicked: function (event) {
+                    console.log(event);
+                }
+            });
+        }
+    );
+
+
+In this code we do three things:
+- We override the ``render`` method to add custom render logic. In this example, we add a div inside our view with a simple text inside it.
+- We declare an event listener to listen to the click events on our view to call the `clicked` method
+- The clicked method will log the DOM click event each time a click is triggered on our view.
+
+Now that we have a view, we need to register it in the ``requirejs.yml`` file
+
+.. code-block:: yaml
+
+    # src/Acme/Bundle/CustomBundle/Resources/config/requirejs.yml
+    config:
+        paths:
+            my_locale_logger: acmecustom/js/locale_logger
+            my_custom_view: acmecustom/js/new_view
+
+
+After cache clearing, asset dump and webpack build, you should be able to use your newly created view.
+
+Create our first form extension
+===============================
+
+Now that we can create a requirejs module and a backbone view, we would like to actually customize the PIM to add our own business logic on top of it.
+
+A little bit of history
++++++++++++++++++++++++
+
+As we are an open source company, our product can used and customized for a lot of different reasons and by a lot of different people (integrators, clients, contributors, technological partners). With this in mind, it was impossible to base our architecture on overrides to customize it.
+
+We decided to create a tree based architecture where each form of the application would be a tree of extensions. If you want to change a part of a page or add something to it, you need to create a requirejs module and register it in the tree of form extensions.
```

Comment:
It's not linked to the reference data, it's linked to the fieldProvider. To override a field, here is the documentation: https://docs.akeneo.com/1.7/cookbook/ui_customization/create_a_new_field_type.html#create-a-field-provider

With this documentation, you can do what you are asking for :)